src/nanoarrow/nanoarrow_device_cuda.c

```diff
@@ -392,6 +417,251 @@ static ArrowErrorCode ArrowDeviceCudaArrayMove(struct ArrowDevice* device_src,
   return ENOTSUP;
 }
 
+static ArrowErrorCode ArrowDeviceCudaResolveBufferSizesAsync(
+    struct ArrowDevice* device, struct ArrowArrayView* array_view, CUstream hstream) {
```

Comment:
`array_view` is host memory here, correct? If so, if the allocated memory is CUDA pinned host memory, then the copies will run truly asynchronously here and return before the memory is actually set to the resolved sizes, but nothing will be blocking the CPU. Is that a user's responsibility in this situation?